cmd/web.go

```diff
@@ -585,12 +585,12 @@ func runWeb(ctx *cli.Context) error {
 			m.Get("/src/*", repo.Home)
 			m.Get("/raw/*", repo.SingleDownload)
 			m.Get("/commits/*", repo.RefCommits)
-			m.Get("/commit/:sha([a-z0-9]{7,40})$", repo.Diff)
+			m.Get("/commit/:sha([a-z0-9]{40})$", repo.Diff)
```

Comment:
This modification don't fix your issue. In fact this route is not used when you call URL like `http://domain.local/zero-x-baadf00d/test/compare/production...production`